system/class/Database/Database.php

```diff
@@ -352,6 +352,10 @@ static function val($value): string
             return $value->getSql();
         }
 
+        if (is_bool($value)) {
+            $value = (int)$value;
```

Comment:
At this point the final value should just be returned, there's no need for it to get processed further in the `is_numeric()` case below.

```suggestion
            return (int) $value;
```